[Admin-Onboarding] New Community Integrations Screen

mzparacha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6025

## Description of Changes
- Added new community integrations screen with Community Directory, Snapshot, Discord (untested), Webhooks, Custom TOS, and Url integrations
- Added mobile responsiveness

## "How We Fixed It"
N/A

## Test Plan
- Enable the `FLAG_NEW_ADMIN_ONBOARDING` flag
- Go to any community where you are an admin
- Go to the `manage/integrations` page
- Verify that the page is responsive on different screen sizes
- Verify that Community Directory integration is working as expected
- Verify that Snapshot integration is working as expected
- Verify that Discord integration is working as expected
- Verify that Webhooks integration is working as expected
- Verify that Custom TOS is working as expected
- Verify that Custom Url "contact us" link opens the device mailbox.
- CA and verify there aren't any issues.

## Deployment Plan
N/A

## Other Considerations
I wasn't able to test discord integration on local properly, because it required some API token's to configure.

mzparacha commented:
If testing on new communities, the admin account might get a page crash. We have a bug report for that https://github.com/hicommonwealth/commonwealth/issues/6366, with a 1-liner fix here https://github.com/hicommonwealth/commonwealth/pull/6367.

masvelio commented:
1. Padding on mobile is definitely too big. According to the new grid (we dont have it yet implemented) padding on mobile should be 16px
<img width="554" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/9e0950e2-24aa-4acd-b625-a4c083887b6f">

2. Hovering over the list items do not highlight the item for some reason. It used to work fine.
3. Sidebar is not reactive if it comes to toggling page directory. If I save changes, directory item should appear/disappear.

https://github.com/hicommonwealth/commonwealth/assets/14819225/3f6b7999-bc9e-4fea-b3b5-6b3ada82e989

Apart from that it looks good, I will take a look on the code shortly

masvelio reviewed (approved):
amazing job! Very clear code, perfect name for files, compoennts, styles. Great structure! 💪

mzparacha commented on packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Directory/Directory.scss:
```diff
@@ -0,0 +1,44 @@
+  width: 100%;
+  gap: 4px;
+
+  .header {
```
Ah... mb. Purged excess css.

mzparacha commented:
Thanks for the awesome review @masvelio. 

1. Updated padding on mobile screens to 16px, it now looks like this

<img width="434" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/32477c55-f644-4f01-885f-8d8c98c3b7a2">

2. Not sure if you mean "highlighting on the text" i.e making it bold? The background tho is highlighted on hover.

https://github.com/hicommonwealth/commonwealth/assets/51641047/a892b879-78d4-4354-88d8-eec5096efc13

3. Forgot to add a `.redraw` emitter. Added now, tested and it works.


Also addressed your other suggestions above. TY!

Israellund reviewed (approved):
LGTM
